### PR TITLE
Ume weighted contrastive loss

### PR DESCRIPTION
**Pull request by karinazad**

### CODE REVIEW

File: src/lobster/model/_ume.py [hunk 1]
```diff
@@ -88,6 +97,8 @@ def __init__(
         beta2: float = 0.98,
         eps: float = 1e-12,
         mask_percentage: float = 0.25,
+        contrastive_loss_weight: float = 1.0,
```
Comment (sjmielke) [inline src/lobster/model/_ume.py:100]: default should be 0.0 (as written in doc)

File: src/lobster/model/_ume.py [hunk 2]
```diff
@@ -418,7 +431,19 @@ def _get_logits_and_labels(self, batch: dict[str, Tensor]) -> tuple[Tensor, Tens
...
+
+        Parameters
+        ----------
+        combined_batch : dict[str, Tensor | list[Modality]]
+            The combined batch to split.
+
+        Returns
+        -------
+        tuple[dict[str, Tensor | list[Modality]], dict[str, Tensor | list[Modality]]]
+            A tuple of two dictionaries, each containing the input IDs and attention masks for the two separate batches.
```
Comment (sjmielke) [inline src/lobster/model/_ume.py:445]: "[...] and individual modalities [...]", right?

**Review by sjmielke — APPROVED** [review]
Small suggestions, though i'd also be curious whether the decision to couple MLM and InfoNCE loss like this will work out well with different batch size requirements for either (CLIP uses like what 32k batch size I think for the contrastive loss, so I don't know if that's also the range we're thinking of for these combined batches---and of course, all of this is assuming we can fit that much in memory without tiling the similarity matrix computation... maybe it's fine for now just to see if it kinda works :) )
